File: backend/internal/models/marketing.go
package models

type MarketingStats struct {
	ID           string  `json:"id"`
	OrgID        string  `json:"org_id"`
	Date         string  `json:"date"`
	CampaignName string  `json:"campaign_name"`
	Spend        float64 `json:"spend"`
	Revenue      float64 `json:"revenue"`
	Impressions  int     `json:"impressions"`
	Clicks       int     `json:"clicks"`
	Conversions  int     `json:"conversions"`
}

type FunnelStep struct {
	Label      string  `json:"label"`
	Value      string  `json:"value"`
	Percentage int     `json:"percentage"`
	SubLabel   string  `json:"subLabel"`
	DropOff    int     `json:"dropOff,omitempty"`
}

type ChannelStats struct {
	ID       string  `json:"id"`
	Name     string  `json:"name"`
	Category string  `json:"category"`
	Spend    float64 `json:"spend"`
	Revenue  float64 `json:"revenue"`
	ROAS     float64 `json:"roas"`
	Icon     string  `json:"icon"`
}

type DashboardInsight struct {
	Title       string `json:"title"`
	Description string `json:"description"`
}

type DashboardData struct {
	Stats    DashboardStats     `json:"stats"`
	Matrix   ROASMatrix         `json:"matrix"`
	Funnel   []FunnelStep       `json:"funnel"`
	Channels []ChannelStats     `json:"channels"`
	Insights []DashboardInsight `json:"insights"`
}

type DailyCVR struct {
	Date    string  `json:"date"`
	Revenue float64 `json:"revenue"`
	Spend   float64 `json:"spend"`
	ROAS    float64 `json:"roas"`
	CTR     float64 `json:"ctr"`
	CVR     float64 `json:"cvr"`
}

type WeeklyROI struct {
	Week       string  `json:"week"`
	AvgRevenue float64 `json:"avg_revenue"`
	AvgSpend   float64 `json:"avg_spend"`
	NetROAS    float64 `json:"net_roas"`
}

type DashboardStats struct {
	Revenue     float64 `json:"revenue"`
	RevenueDiff float64 `json:"revenue_diff"`
	Spend       float64 `json:"spend"`
	SpendDiff   float64 `json:"spend_diff"`
	ROAS        float64 `json:"roas"`
	ROASDiff    float64 `json:"roas_diff"`
	Conversions int     `json:"conversions"`
	ConvDiff    float64 `json:"conv_diff"`
}

type ROASMatrixCell struct {
	DayOfWeek int     `json:"day_of_week"` // 0-6 (Sun-Sat)
	HourOfDay int     `json:"hour_of_day"` // 0-23
	ROAS      float64 `json:"roas"`
}

type ROASMatrix []ROASMatrixCell

File: frontend/src/types/marketing.ts
export interface MarketingData {
  id: string;
  date: string;
  spend: number;
  revenue: number;
  roas: number;
}

export interface KPI {
  title: string;
  value: string;
  trendIcon: string;
  trendValue: string;
  trendBgClass: string;
  trendTextClass: string;
  barClasses?: string[];
}

export interface FunnelLayerData {
  label: string;
  value: string;
  metricLabel: string;
  metricValue: string;
  dropValue?: string;
  colorClassMain: string;
  colorClassHover: string;
  widthClass: string;
  textMainClass: string;
  textSubClass: string;
}

export interface InsightData {
  type: "success" | "warning" | "info";
  title: string;
  description: string;
  icon: string;
}

export interface ChannelPerformance {
  id: string;
  name: string;
  category: string;
  spend: string;
  revenue: string;
  roas: string;
  icon: string;
  color: "indigo" | "orange" | "green" | "slate";
}

export interface ROASMatrixCell {
  day_of_week: number;
  hour_of_day: number;
  roas: number;
}

export interface FunnelStep {
  label: string;
  value: string;
  percentage: number;
  subLabel: string;
  dropOff?: number;
}

export interface ChannelStats {
  id: string;
  name: string;
  category: string;
  spend: number;
  revenue: number;
  roas: number;
  icon: string;
}

export interface DashboardInsight {
  title: string;
  description: string;
}

export interface DashboardStats {
  revenue: number;
  revenue_diff: number;
  spend: number;
  spend_diff: number;
  roas: number;
  roas_diff: number;
  conversions: number;
  conv_diff: number;
}

export interface DashboardData {
  stats: DashboardStats;
  kpis: KPI[];
  matrix: ROASMatrixCell[];
  funnel: FunnelStep[];
  channels: ChannelStats[];
  insights: DashboardInsight[];
}

export interface PivotData {
  date: string;
  actual: number | null;
  predicted: number | null;
}

export interface PivotDetails {
  id: string;
  pivotDate: string;
  status: "achieved" | "deviated";
  memo: {
    intent: string;
    diagnosis: string[];
    conclusion: string;
  };
  timeline: PivotData[];
}

export interface LineageTelemetry {
  source: string;
  confidence: string;
  engine?: string;
  timestamp?: string;
  zScore?: number;
  sourceTable?: string;
  executionSql?: string;
  researchNotes?: string;
}

export interface Lineage {
  source: string;
  query_id?: string;
  sql_ref?: string;
  timestamp?: string;
}

export interface DailyCVR {
  date: string;
  revenue: number;
  spend: number;
  roas: number;
  ctr: number;
  cvr: number;
  is_anomaly?: boolean;
  z_score?: number;
  lineage?: Lineage;
}

export interface ResponseMetadata {
  engine: string;
  confidence: string;
  sql_ref?: string;
}

